src/cpack.ts

```diff
@@ -0,0 +1,142 @@
+    readonly onPackagingEnabledChanged = this.packagingEnabledEmitter.event;
+
+    dispose() {
+        this.packagingEnabledEmitter.dispose();
+    }
+
+    private async getCPackArgs(driver: CMakeDriver, packagePreset?: PackagePreset): Promise<string[] | undefined> {
+        let cpackArgs: string[] = [];
+        if (!packagePreset && driver.packagePreset) {
+            packagePreset = driver.packagePreset;
+        }
+        if (!driver.useCMakePresets || !packagePreset) {
+            return undefined;
+        }
+
+        const opts = driver.expansionOptions;
+        const args = [];
+        for (const value of this.ws.config.cpackArgs) {
+            args.push(await expandString(value, opts));
+        }
+
+        // Note: in CMake Tools, we don't run cmake or cpack with --preset argument. We generate the equivalent command line from all the properties
+        cpackArgs = [];
+        if (packagePreset.vendorName) {
+            cpackArgs.push("--vendor", `${packagePreset.vendorName}`);
+        }
+        if (packagePreset.generators?.length) {
+            cpackArgs.push("-G", `${packagePreset.generators.join(";")}`);
+        }
+        if (packagePreset.configurations?.length) {
+            cpackArgs.push("-C", `${packagePreset.configurations.join(";")}`);
+        }
+        if (packagePreset.configFile) {
+            cpackArgs.push("--config", `${packagePreset.configFile}`);
+        }
+        if (packagePreset.output?.debug) {
+            cpackArgs.push("--debug");
+        }
+        if (packagePreset.output?.verbose) {
+            cpackArgs.push("--verbose");
+        }
+        if (packagePreset.packageName) {
+            cpackArgs.push("-P", `${packagePreset.packageName}`);
+        }
+        if (packagePreset.packageVersion) {
+            cpackArgs.push("-R", `${packagePreset.packageVersion}`);
+        }
+        if (packagePreset.packageDirectory) {
+            cpackArgs.push("-B", `${packagePreset.packageDirectory}`);
+        }
+
+        if (packagePreset.variables) {
+            for (const varName in packagePreset.variables) {
+                cpackArgs.push(`-D ${varName}=${packagePreset.variables[varName]}`);
+            }
+        }
+
+        cpackArgs = cpackArgs.concat(args);
+        return cpackArgs;
+    }
+
+    public async runCPack(driver: CMakeDriver, packagePreset?: PackagePreset, consumer?: proc.OutputConsumer): Promise<number> {
+        const cpackpath = await this.ws.getCPackPath(driver.cmakePathFromPreset);
+        if (cpackpath === null) {
+            log.info(localize('cpack.path.not.set', 'CPath path is not set'));
+            return -2;
+        }
+
+        let cpackArgs: string[];
+        if (driver.useCMakePresets && !driver.packagePreset) {
+            log.error(localize('package.preset.not.set', 'Package preset is not set'));
+            return -3;
+        } else {
+            const opts = driver.expansionOptions;
```

Comment:
Ah, this code makes it make more sense why `Run CPack` is the better command name, please ignore my other comment about possibly renaming that command.